core/store/src/trie/from_flat.rs

```diff
@@ -0,0 +1,49 @@
+use crate::flat::{store_helper, FlatStorageManager};
+use crate::{ShardTries, Store, Trie, TrieConfig, TrieDBStorage, TrieStorage};
+use near_primitives::{shard_layout::ShardUId, state::FlatStateValue};
+use std::time::Instant;
+
+pub fn construct_trie_from_flat(store: Store, write_store: Store, shard_uid: ShardUId) {
+    let timer = Instant::now();
+    let trie_storage = TrieDBStorage::new(store.clone(), shard_uid);
+    let mut num_entries = 0;
+    let entries =
+        store_helper::iter_flat_state_entries(shard_uid, &store, None, None).map(|entry| {
```

Comment:
TODO: Probably need to break this up into separate iterations as we can't handle the whole trie in-memory in one go.

Reply:
Sounds good, the iter_flat_state_entries method already returns an iterator so you should be able to easily batch it. Depending on the implementation of Trie::update and if it does any optimizations or not, the batching strategy may actually change performance so be mindful of that please.

Reply:
Done